tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,544 @@
+{
+	LOG_DBG("VOCS state callback");
+}
+
+static void vocs_location_cb(struct bt_vocs *inst, int err, uint32_t location)
+{
+	LOG_DBG("VOCS location callback");
+}
+
+static void vocs_description_cb(struct bt_vocs *inst, int err,
+				char *description)
+{
+	LOG_DBG("VOCS desctripion callback");
+}
+
+static struct bt_vocs_cb vocs_cb = {
+	.state = vocs_state_cb,
+	.location = vocs_location_cb,
+	.description = vocs_description_cb
+};
+
+void vocs_audio_desc(const char *description, uint8_t len)
+{
+	int err;
+
+	LOG_DBG("VOCS description");
+	for (int i = 0; i < (int)CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_vocs_description_set(included.vocs[i], description);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_VOCS, VOCS_UPDATE_DESC,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		}
+	}
+}
+
+void vocs_audio_loc(uint8_t *data, uint8_t len)
+{
+	int err;
+
+	LOG_DBG("VOCS location");
+	for (int i = 0; i < (int)CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_vocs_location_set(included.vocs[i], (int)data);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_VOCS, VOCS_UPDATE_LOC,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		}
+	}
+}
+
+void tester_handle_vocs(uint8_t opcode, uint8_t index, uint8_t *data,
+			uint16_t len)
+{
+	switch (opcode) {
+	case VOCS_INIT:
+		vcp_init();
+		break;
+	case VOCS_READ_SUPPORTED_COMMANDS:
+		vocs_supported_commands(data,len);
+		break;
+	case VOCS_UPDATE_DESC:
+		vocs_audio_desc(data, len);
+		break;
+	case VOCS_UPDATE_LOC:
+		vocs_audio_loc(data, len);
+		break;
+	default:
+		tester_rsp(BTP_SERVICE_ID_VOCS, opcode, index,
+				BTP_STATUS_UNKNOWN_CMD);
+		break;
+	}
+}
+
+/* General profile handling */
+static void set_register_params(uint8_t gain_mode)
+{
+	char input_desc[CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT][16];
+	char output_desc[CONFIG_BT_VCP_VOL_REND_VOCS_INSTANCE_COUNT][16];
+
+	memset(&vcp_register_param, 0, sizeof(vcp_register_param));
+
+	for (int i = 0; i < ARRAY_SIZE(vcp_register_param.vocs_param); i++) {
+		vcp_register_param.vocs_param[i].location_writable = true;
+		vcp_register_param.vocs_param[i].desc_writable = true;
+		vcp_register_param.vocs_param[i].output_desc = output_desc[i];
+		vcp_register_param.vocs_param[i].cb = &vocs_cb;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(vcp_register_param.aics_param); i++) {
+		vcp_register_param.aics_param[i].desc_writable = true;
+		vcp_register_param.aics_param[i].description = input_desc[i];
+		vcp_register_param.aics_param[i].type = BT_AICS_INPUT_TYPE_DIGITAL;
+		vcp_register_param.aics_param[i].status = g_aics_active;
+		vcp_register_param.aics_param[i].gain_mode = gain_mode;
+		vcp_register_param.aics_param[i].units = 1;
+		vcp_register_param.aics_param[i].min_gain = 0;
+		vcp_register_param.aics_param[i].max_gain = 100;
+		vcp_register_param.aics_param[i].cb = &aics_cb;
+	}
+
+	/* Default values */
+	vcp_register_param.step = 1;
+	vcp_register_param.mute = BT_VCP_STATE_UNMUTED;
+	vcp_register_param.volume = 100;
```

Comment:
Aren't these values also set in `vcp_init`?

Reply:
There is now only one place that sets them.